### PR TITLE
CloudWatch: Fix `all-log-groups` endpoint when called without logGroupNamePrefix 

### PR DESCRIPTION
**Pull request by lionralfs**
**What this PR does / why we need it**:

As discussed in #56185 (specifically, [this comment](https://github.com/grafana/grafana/pull/56185#discussion_r987268832)), the `/all-log-groups` endpoint currently returns the following error when either passing an empty string or not passing a `logGroupNamePrefix` query parameter at all:

```
unexpected error InvalidParameter: 1 validation error(s) found.
- minimum field size of 1, DescribeLogGroupsInput.LogGroupNamePrefix.
```

This PR fixes that by checking the string length before passing it to the AWS client.

### CODE REVIEW

File: pkg/tsdb/cloudwatch/cloudwatch_test.go [hunk 1]
```diff
@@ -327,6 +327,94 @@ func TestQuery_ResourceRequest_DescribeAllLogGroups(t *testing.T) {
...
+
+		expectedFirst := &cloudwatchlogs.DescribeLogGroupsInput{
+			Limit: aws.Int64(50),
+		}
+		expectedSecond := &cloudwatchlogs.DescribeLogGroupsInput{
+			Limit:     aws.Int64(50),
+			NextToken: aws.String("foo"),
+		}
+		assert.Equal(t, expectedFirst, firstCall)
+		assert.Equal(t, expectedSecond, secondCall)
```
Comment (fridgepoet) [inline pkg/tsdb/cloudwatch/cloudwatch_test.go:342]:
```suggestion
```

I think we don't need these assertions, what do you think? Apologies if I'm missing something that you wanted to show here.;

**Review by fridgepoet — APPROVED** [review]
Great catch! Thank you!

File: pkg/tsdb/cloudwatch/cloudwatch_test.go (hunk 1 above)
Comment (lionralfs) [reply pkg/tsdb/cloudwatch/cloudwatch_test.go:342]: This test case tries to fetch multiple "pages", so I thought it makes sense to assert that the "NextToken" from the first response is passed in the second request. You're right in that this modification doesn't really fit for what this PR is trying to achieve.

Comment (fridgepoet) [reply pkg/tsdb/cloudwatch/cloudwatch_test.go:342]: If that's ok with you, let's leave that out for now and merge in everything else you have here.

Comment (lionralfs) [reply pkg/tsdb/cloudwatch/cloudwatch_test.go]: Sure!

**Comment by fridgepoet** [comment]
@lionralfs can you pull in changes from `main` please?

**Comment by lionralfs** [comment]
@fridgepoet done :)

File: pkg/tsdb/cloudwatch/cloudwatch_test.go [hunk 2]
```diff
@@ -330,6 +330,81 @@ func TestQuery_ResourceRequest_DescribeAllLogGroups(t *testing.T) {
...
+	t.Run("Should call api without LogGroupNamePrefix when an empty string is passed in resource call", func(t *testing.T) {
+		cli = fakeCWLogsClient{
+			logGroups: []cloudwatchlogs.DescribeLogGroupsOutput{
+				{LogGroups: []*cloudwatchlogs.LogGroup{}},
+			},
+		}
+
+		im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+			return datasourceInfo{}, nil
+		})
```
Comment (fridgepoet) [inline pkg/tsdb/cloudwatch/cloudwatch_test.go:381]:
```suggestion
```
same as above

**Review by fridgepoet — APPROVED** [review]

File: pkg/tsdb/cloudwatch/cloudwatch_test.go [hunk 3]
```diff
@@ -330,6 +330,81 @@ func TestQuery_ResourceRequest_DescribeAllLogGroups(t *testing.T) {
...
+	t.Run("Should call api with LogGroupNamePrefix if passed in resource call", func(t *testing.T) {
+		cli = fakeCWLogsClient{
+			logGroups: []cloudwatchlogs.DescribeLogGroupsOutput{
+				{LogGroups: []*cloudwatchlogs.LogGroup{}},
+			},
+		}
+
+		im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+			return datasourceInfo{}, nil
+		})
```
[the remark this answers is not in the document]
Comment (lionralfs) [reply pkg/tsdb/cloudwatch/cloudwatch_test.go:343]: Oh, sorry. Didn't check for failing tests after pulling in the upstream changes. I'll commit your suggestions

**Review by iwysiu — APPROVED** [review]

**Pull request merged**